homeassistant/components/nanoleaf/light.py

```diff
@@ -171,27 +172,39 @@ def supported_features(self):
 
     def turn_on(self, **kwargs):
         """Instruct the light to turn on."""
-        self._light.on = True
         brightness = kwargs.get(ATTR_BRIGHTNESS)
         hs_color = kwargs.get(ATTR_HS_COLOR)
         color_temp_mired = kwargs.get(ATTR_COLOR_TEMP)
         effect = kwargs.get(ATTR_EFFECT)
+        transition = kwargs.get(ATTR_TRANSITION)
 
         if hs_color:
             hue, saturation = hs_color
             self._light.hue = int(hue)
             self._light.saturation = int(saturation)
 
+        if transition:
+            if brightness:  # tune to the required brightness in n seconds
+                self._light.brightness_transition(
+                    int(brightness / 2.55), int(transition))
+            else:  # If brightness is not specified, assume full brightness
+                self._light.brightness_transition(100, int(transition))
+        else:  # If no transition is occurring, turn on the light
+            self._light.on = True
+            if brightness:
+                self._light.brightness = int(brightness / 2.55)
         if color_temp_mired:
             self._light.color_temperature = mired_to_kelvin(color_temp_mired)
```

Comment:
Should the color temperature setting be moved above the transition (like HS color is) so it is set as the light turns on?

Reply:
Yeah, thanks for catching that! It should not (tm) be very noticeable because turning on starts off very dim, but nevertheless should look better

Reply:
Okay. I didn't know if it would abort the transition.

Approved now, will be merged when the automatic tests complete.